Sources/_StringProcessing/Compiler.swift

```diff
@@ -79,9 +79,17 @@ class Compiler {
     case .trivia, .empty:
       break
 
-    // FIXME: This can't be right...
     case .group(let g):
-      try emit(g.child)
+      switch g.kind.value {
+
+      case .lookahead, .negativeLookahead,
+          .lookbehind, .negativeLookbehind:
+        try compileLookaround(g.kind.value, g.child)
```

Comment:
We could use a `Lookaround` enum in the AST to group these four cases. Then `compileLookaround(_:_:)` would not need to handle unreachable cases.

Reply:
Added a `var lookaroundKind: (forwards: Bool, positive: Bool)?` to `Group`. This is probably a little more useful than an enum, which we'd want to immediately destructure or add these APIs to anyways. If we have more uses for more semantic info here, we can make the enum.